lib/Parser/Hash.h

```diff
@@ -266,8 +267,16 @@ struct Ident
                     // Without parameter scope, we would have just pushed the ref instead of inserting.
                     // We effectively had a false positive match (a parameter scope ref with no sym)
                     // so we need to push the Pid rather than inserting.
-                    newRef->prev = m_pidRefStack;
-                    m_pidRefStack = newRef;
+                    if (prevRef)
+                    {
+                        newRef->prev = prevRef->prev;
+                        prevRef->prev = newRef;
+                    }
+                    else
+                    {
+                        newRef->prev = m_pidRefStack;
```

Comment:
We were talking offline about how we weren't sure if this case would come about with the above condition. Do you have an example that breaks our assumption?